sklearn/utils/validation.py

```diff
@@ -1259,3 +1270,30 @@ def inner_f(*args, **kwargs):
         kwargs.update({k: arg for k, arg in zip(all_args, args)})
         return f(**kwargs)
     return inner_f
+
+
+def _check_fit_params(fit_params):
+    """Check and validate the parameters passed during `fit`.
+
+    Parameters
+    ----------
+    fit_params : dict
+        Dictionary containing the parameters passed at fit.
+
+    Returns
+    -------
+    fit_params_validated : dict
+        Validated parameters. We ensure that the values support indexing.
+    """
+    fit_params_validated = {}
+    for param_key, param_value in fit_params.items():
+        if not _is_arraylike(param_value):
```

Comment:
Reading the code again, in `_index_param_value` which is called by `_fit_and_score` / `_fit_and_predict`, downstream in the `Parallel` loop, we do:

```python
def _index_param_value(X, v, indices):
    """Private helper function for parameter value indexing."""
    if not _is_arraylike(v) or _num_samples(v) != _num_samples(X):
        # pass through: skip indexing
        return v
    if sp.issparse(v):
        v = v.tocsr()
    return _safe_indexing(v, indices)
```

Maybe we should use the exact same condition for now for consistency:

```python
    if not _is_arraylike(v) or _num_samples(v) != _num_samples(X):
        # pass through: skip indexing
        fit_params_validated[param_key] = param_value
```

I know this pattern is a bit hackish but I think we should preserve it for now for the sake of backward compat.

Reply:
Also the lines:

```python
    if sp.issparse(v):
        v = v.tocsr()
```

in `_index_param_value` are now redundant. The `tocsr()` conversion has already happened in the new `_check_fit_params`: let's remove those 2 lines from `_index_param_value`.